static/settings.ts

```diff
@@ -159,7 +159,7 @@ class Slider extends BaseSetting {
     }
 
     override getUi(): number {
-        return parseInt(this.val()?.toString() ?? '0');
+        return parseInt(this.val().toString());
```

Comment:
This one is a TS-side error probably: 
https://github.com/compiler-explorer/compiler-explorer/pull/3400 fixed an issue with embeded links not working because val() returned undefined.
This happens when the setting element is not found, which is the case when in embed mode, because no settings html is actually created.

I think the better option would be to specify that val can in fact return undefined and handle the cases that do not account for that (Even though the only place that suffered was this one!)

Reply:
Thanks for spotting this! I've updated .val() to indicate it can return undefined, definitely the better way to go!

Reply:
Can you also bring back this check? :)

Reply:
Yeah brought back this check (and added one for the use of .val() on line 186)

Reply:
Oops sorry, the "Files changed" view was not updating there for a second